### PR TITLE
Meta changes

### PR DESCRIPTION
**Pull request by xen-0**
Refactor meta subclass registration and add initial NumberMeta implementation

### CODE REVIEW

File: malcolm/core/attributemeta.py [hunk 1]
```diff
@@ -35,15 +35,17 @@ def to_dict(self):
         return d
 
     @classmethod
-    def register_subclass(cls, subcls, metaOf):
+    def register_subclass(cls, metaOf, *args):
         """Register a subclass so from_dict() works
 
         Args:
             subcls (AttributeMeta): AttributeMeta subclass to register
```
Comment (thomascobb) [inline malcolm/core/attributemeta.py]: docs need updating too...

File: malcolm/core/numbermeta.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from malcolm.core.attributemeta import AttributeMeta
+
+from pkg_resources import require
+require("numpy")
+import numpy as np
+
+
+_metas = {
```
Comment (thomascobb) [inline malcolm/core/numbermeta.py]: (matter of taste) I probably wouldn't use an intermediate dictionary, I think it's probably easier to read if you just put the strings straight into the decorator calls...

File: malcolm/core/numbermeta.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+@AttributeMeta.register_subclass(_metas[np.int32], np.int32)
+@AttributeMeta.register_subclass(_metas[np.float64], np.float64)
+class NumberMeta(AttributeMeta):
+    """Meta object containing information for a numerical value"""
+
+    def __init__(self, name, description, dtype):
+        super(NumberMeta, self).__init__(name=name, description=description)
+        self.dtype = dtype
+        # TODO where and when should this be set?
```
Comment (thomascobb) [inline malcolm/core/numbermeta.py]: I think the base class should do this. If we maintain the inverse mapping from (subcls, args) -> metaOf is AttributeMeta.register_subclass, then AttributeMeta.__init__(name, description, *args) can set self.metaOf based on this information

File: malcolm/core/numbermeta.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from malcolm.core.attributemeta import AttributeMeta
+from malcolm.compat import base_string
+
+from pkg_resources import require
```
Comment (thomascobb) [inline malcolm/core/numbermeta.py]: No requires in the source code, just in the test entry points please...

**Comment by xen-0** [comment]
Ok, so this pull-request is starting to include a lot of tangential changes...

**Pull request merged**